test/tls12groupselection_test.c

```diff
@@ -0,0 +1,201 @@
+#include "testutil.h"
+#include "helpers/ssltestlib.h"
+#include <openssl/objects.h>
+
+#define TEST_true_or_end(a) if (!TEST_true(a)) \
+        goto end;
+
+#define SERVER_PREFERENCE 1
+#define CLIENT_PREFERENCE 0
+
+#define WORK_ON_SSL_OBJECT 1
+#define WORK_ON_CONTEXT 0
+
+#if !defined(OPENSSL_NO_EC) && !defined(OPENSSL_NO_TLS1_2)
```

Comment:
Not necessary if you make the build.info change I suggest. The test recipe will never call us anyway if these things are disabled.